Bjs2 65888 Подписка на других пользователей

Kuruzo opened the pull request.

Kuruzo committed:
init SubscriptionController SubscriptionService

Kuruzo committed:
init DataValidationException

Kuruzo committed:
fitch user subscription

Kuruzo committed:
clear import

Kuruzo committed:
unsubscribe feature

Kuruzo committed:
init UserDto

Kuruzo committed:
edit UserDto

Kuruzo committed:
edit UserDto

Kuruzo committed:
init UserFilterDto

Kuruzo committed:
add UserFilter impl Predicate<>

Kuruzo committed:
add getFollowers feature

Kuruzo committed:
refactor UserFilter

Kuruzo committed:
init User & UserFilter mappers

Kuruzo committed:
refactor controller with mappers

Kuruzo committed:
BJS2-65888

Kuruzo committed:
init dev branch

Kuruzo committed:
init BJS2-65888 branch

romankondratev commented on src/main/java/school/faang/user_service/UserServiceApplication.java:
```diff
@@ -22,5 +22,6 @@ public ObjectMapper objectMapper() {
         objectMapper.registerModule(new JavaTimeModule());
         objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
         return objectMapper;
+
```
А зачем пустая строка?

Kuruzo replied:
Мисс клик. Прошёлся по коду. вроде все лишнее убрал)

KAS1990 commented on src/main/java/school/faang/user_service/exception/DataValidationException.java:
```diff
@@ -0,0 +1,8 @@
+package school.faang.user_service.exception;
+
+public class DataValidationException extends IllegalArgumentException {
```
Зачем ты создал этот класс, когда есть стандартный org.junit.jupiter.params.shadow.com.univocity.parsers.common.DataValidationException

Kuruzo replied:
Почитал про org.junit.jupiter.params.shadow.com.univocity.parsers.common  Это исключение про парсинг у меня всё же бизнес логика и наследуется от илегаларгумент говоря дополнительно что такие данные не соответствуют ожидаемым

Kuruzo committed:
check code style

romankondratev reviewed (approved):
Отправляй на ревью Руслану =)

Kuruzo committed:
create WorkScheduleDto

Kuruzo committed:
create WorkScheduleMapper

Kuruzo committed:
refactor WorkScheduleDto

Kuruzo committed:
User.workSchedule: WorkSchedule > List<>
create EntityAlreadyExistsException
controller/service.WorkSchedule add: addWorkSchedule,updateWorkSchedule,getWorkScheduleById
